src/features/events/components/EventPopper/MultiEventPopper/SingleEvent.tsx

```diff
@@ -0,0 +1,286 @@
+import { makeStyles } from '@mui/styles';
+import NextLink from 'next/link';
+import { useMessages } from 'core/i18n';
+import { useRouter } from 'next/router';
+import {
+  AccessTime,
+  ArrowForward,
+  EmojiPeople,
+  FaceOutlined,
+  MailOutline,
+  People,
+  PlaceOutlined,
+} from '@mui/icons-material';
+import { Box, Button, Checkbox, Link, Typography } from '@mui/material';
+import { FC, useContext } from 'react';
+
+import LocationName from '../../LocationName';
+import messageIds from 'features/events/l10n/messageIds';
+import Quota from '../Quota';
+import StatusDot from '../StatusDot';
+import useModel from 'core/useModel';
+import { ZetkinEvent } from 'utils/types/zetkin';
+import { ZUIConfirmDialogContext } from 'zui/ZUIConfirmDialogProvider';
+import ZUIEllipsisMenu from 'zui/ZUIEllipsisMenu';
+import ZUIIconLabel from 'zui/ZUIIconLabel';
+import ZUIPerson from 'zui/ZUIPerson';
+import ZUIPersonHoverCard from 'zui/ZUIPersonHoverCard';
+import ZUITimeSpan from 'zui/ZUITimeSpan';
+import EventDataModel, {
+  EventState,
+} from 'features/events/models/EventDataModel';
+
+const useStyles = makeStyles(() => ({
+  description: {
+    '-webkit-box-orient': 'vertical',
+    '-webkit-line-clamp': 3,
+    display: '-webkit-box',
+    maxHeight: '100%',
+    overflow: 'hidden',
+    whiteSpace: 'normal',
+    width: '100%',
+  },
+}));
+
+interface SingleEventProps {
+  event: ZetkinEvent;
+  onClickAway: () => void;
```

Comment:
I would call this `onClose`, because `onClickAway` is so tightly connected to the idea of a `ClickAwayListener` and this is not really that, but just a regular close triggered by a number of different things. But that's just a matter of preference.